lib/cocoapods/installer/analyzer.rb

```diff
@@ -554,11 +563,20 @@ def filter_pod_targets_for_target_definition(target_definition, pod_targets_by_t
             whitelists = dependencies.map do |dependency|
               target_definition.pod_whitelisted_for_configuration?(dependency.name, configuration_name)
             end.uniq
+            if resolver_spec.transitive?
+              # Include the whitelist config for the source of the transitive dependency, so we can exclude
+              # the transitive dependencies of whitelisted dependencies
+              whitelists += resolver_spec.dependent_specs.map do |dependent_spec|
```

Comment:
I don't think this check is correct, since a spec can come in transitively through multiple different specs, and we need to make sure we're only merging things where the only specs depending on that spec have been excluded

e.g.

A -> B
C -> B

A whitelisted, C not, we still need to include B

Reply:
good catch, definitely need to address this

Reply:
also uncovered a case where the subspec of a pod depends on the current pod, it will always be whitelisted. this needs to filter `dependent_specs` to only include the target definition's direct dependencies which have been whitelisted for the current configuration